internal/character/march7th/talent.go

```diff
@@ -0,0 +1,141 @@
+package march7th
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Talent             = "march7th-talent"
+	TalentCount        = "march7th-talentcount"
+	MarchAllyMark      = "march7th-shield-counter"
+	MarchCounterMark   = "march7th-counter-mark"
+	MarchCounterAttack = "march7th-counter"
+)
+
+/*
+Note:
+High level breakdown of how March's talent works, as far as I can tell from the DM:
+When March joins batttle, she is given a modifier that represents her actual talent.
+This modifier, when added, adds another modifier to all of March's teammates, which I called MarchAllyMark
+MarchAllyMark listens for whenever the person it is attached to is about to be attacked. If at least one of the targets in
+that attack is shielded, AND the attacker is an enenmy, add a modifier called MarchCounterMark to the attacker
+MarchCounterMark listens for when whoever it is attached to is done attacking.
+Afterwards, it tells March to counter attack. If successful, March will counter, and (?) remove
+one stack from the modifier that determines if march can counter or not
+MarchCounterMark will not be added to an enemy if March does not have enough counters left.
+March regenerates her counters on Phase2End.
+*/
```

Comment:
Instead of using modifiers, majority of this logic is a better fit for subscriptions. A subscription runs on every event of that type rather than attached to a specific target. For cases like the "MarchAllyMark", this allows you to get every `AttackStart` event and then filter down to only the ones you care about (unlike the game, we can write code and do not have to be limited to modifiers).

Thinking out loud on how to implement this talent:
* `AttackStart` subscription that does the logic of `MAvatar_March7th_00_BPSkill_Shield_Counter` (will need to loop through list of targets)
* Can keep counter as either a modifier or variable in the char struct. A variable is easier to implement, a modifier is better for data traceability (char variables are not logged so harder for sim users to trace)
* Add mark modifier to enemies from the `AttackStart` subscription which will insert the counter `OnAfterAttack` (equivalent of `MAvatar_March7th_00_BPSkill_Shield_Mark`)
* Can still have talent modifier with `OnPhase2` subscription to reset stacks (need to also make sure that character starts with stacks)